File: modules/rest/src/main/java/dev/httpmarco/polocloud/modules/rest/auth/user/permission/Permission.java
package dev.httpmarco.polocloud.modules.rest.auth.user.permission;

import lombok.Getter;
import lombok.experimental.Accessors;

import java.util.ArrayList;
import java.util.List;

@Getter
@Accessors(fluent = true)
public class Permission {

    private final List<String> permissions;

    public Permission() {
        this.permissions = new ArrayList<>();
    }

    public void addPermission(String permission) {
        this.permissions.add(permission);
    }

    public void removePermission(String permission) {
        this.permissions.remove(permission);
    }

    public boolean hasPermission(String permission) {
        if (this.permissions.contains("*")) return true;
        return this.permissions.contains(permission);
    }
}

File: modules/rest/src/main/kotlin/dev/httpmarco/polocloud/modules/rest/auth/user/permission/Permission.kt
package dev.httpmarco.polocloud.modules.rest.auth.user.permission

open class Permission {

    val permissions: MutableList<String> = mutableListOf()

    fun addPermission(permission: String) {
        if (permission.isNotBlank() && !this.permissions.contains(permission)) {
            this.permissions.add(permission)
        }
    }

    fun addPermissions(vararg permissions: String) {
        for (permission in permissions) {
            addPermission(permission)
        }
    }

    fun removePermission(permission: String) {
        this.permissions.remove(permission)
    }

    fun hasPermission(permission: String): Boolean {
        return "*" in this.permissions || permission in this.permissions
    }
}